File: app/src/main/java/com/jml/rappichallenge/repository/api/MdbRoutes.java
package com.jml.rappichallenge.repository.api;

import com.jml.rappichallenge.models.enums.Language;
import com.jml.rappichallenge.models.enums.Sorting;

import androidx.annotation.StringRes;

class MdbRoutes
{
	static class Version
	{
		static final String V3 = "3";
	}

	static  class Path
	{
		static final String Discover = "discover";
		static final String Movie = "movie";
	}

	static class QKey
	{
		static final String ApiKey = "api_key";
		static final String Language = "language";
		static final String Sorting = "sort_by";
		static final String IncludeAdult = "include_adult";
		static final String IncludeVideo = "include_video";
		static final String Page = "page";
		static final String Keyword = "with_keywords";
	}

	static class LanguageCode
	{
		private static final String EN_code = "en-US";
		private static final String ES_code = "es-ES";

		static String fromLanguage(Language language)
		{
			switch (language)
			{
				case English:
					return EN_code;
				case Spanish:
					return ES_code;
				default:
						return EN_code;
			}
		}
	}

	static class SortingCode
	{
		private static final String PopulariteCode = "popularity.desc";
		private static final String RatingCode = "vote_average.desc";
		private static final String DateCode = "primary_release_date.desc";

		static String fromSorting(Sorting sorting)
		{
			switch (sorting)
			{
				case Popularity:
					return PopulariteCode;
				case Rating:
					return RatingCode;
				case Date:
					return DateCode;
				default:
					return PopulariteCode;
			}
		}
	}

	static final String RootUrl = "https://api.themoviedb.org/";
}

File: app/src/main/java/com/jml/rappichallenge/repository/api/MdbRoutes.kt
package com.jml.rappichallenge.repository.api

import com.jml.rappichallenge.models.enums.Language
import com.jml.rappichallenge.models.enums.Sorting

internal object MdbRoutes {

    val RootUrl = "https://api.themoviedb.org/"

    internal object Version {
        const val V3 = "3"
    }

    internal object Path {
        const val Discover = "discover"
        const val Popular = "popular"
        const val TopRated = "top_rated"
        const val Upcoming = "upcoming"
        const val Movie = "movie"
    }

    internal object QKey {
        const val ApiKey = "api_key"
        const val Language = "language"
        const val Sorting = "sort_by"
        const val IncludeAdult = "include_adult"
        const val IncludeVideo = "include_video"
        const val Page = "page"
        const val Keyword = "with_keywords"
    }

    internal object LanguageCode {
        private const val EN_code = "en-US"
        private const val ES_code = "es-ES"

        fun fromLanguage(language: Language): String {
            when (language) {
                Language.English -> return EN_code
                Language.Spanish -> return ES_code
                else -> return EN_code
            }
        }
    }

    internal object SortingCode {
        private const val PopulariteCode = "popularity.desc"
        private const val RatingCode = "vote_average.desc"
        private const val DateCode = "primary_release_date.desc"

        fun fromSorting(sorting: Sorting): String {
            when (sorting) {
                Sorting.Popularity -> return PopulariteCode
                Sorting.Rating -> return RatingCode
                Sorting.Date -> return DateCode
                else -> return PopulariteCode
            }
        }
    }
}
